src/background/privacySettings.js

```diff
@@ -0,0 +1,26 @@
+import {Component} from "./component.js";
+import {Logger} from "./logger.js";
+
+const log = Logger.logger("PrivacySettings");
+
+// Make sure we have dFPI (or stronger) enabled.
+let enforceCookieIsolation = async () => {
+  const initialConfig = await browser.privacy.websites.cookieConfig.get({});
+  log(`initial privacy config: ${initialConfig}`);
+
+  if (initialConfig.value.behavior === "reject_trackers") {
+    // Cookie behavior is too weak. Let's upgrade to dFPI.
+    await browser.privacy.websites.cookieConfig.set({
+      value: {behavior: "reject_trackers_and_partition_foreign"}});
+  }
+
+  const finalConfig = await browser.privacy.websites.cookieConfig.get({});
+  log(`final privacy config: ${finalConfig}`);
+};
+
+export class PrivacySettings extends Component {
+  constructor(receiver) {
+    super(receiver);
+    browser.runtime.onInstalled.addListener(enforceCookieIsolation);
```

Comment:
I see 3 options here:

1. Should we enable the reject_trackers_and_partition_foreign only when the proxy is active? If yes, we should move this code in setProxyState().
2. We should set this setting always, not only at the first installation. If yes, this code should go to the init().
3. As you wrote, just at the first installation.

Can you write a comment about why 3? I vote for 1 or 2 :)

Reply:
Thanks for the review! I agree that 1 is better. I pushed a fixup commit to the pull request.